components/core/src/clp/streaming_archive/reader/Segment.cpp

```diff
@@ -33,38 +33,19 @@ ErrorCode Segment::try_open(string const& segment_dir_path, segment_id_t segment
         return ErrorCode_Success;
     }
 
-    // Get the size of the compressed segment file
-    boost::system::error_code boost_error_code;
-    size_t segment_file_size = boost::filesystem::file_size(segment_path, boost_error_code);
-    if (boost_error_code) {
-        SPDLOG_ERROR(
-                "streaming_archive::reader::Segment: Unable to obtain file size for segment: "
-                "{}",
-                segment_path.c_str()
-        );
-        SPDLOG_ERROR("streaming_archive::reader::Segment: {}", boost_error_code.message().c_str());
-        return ErrorCode_Failure;
-    }
-
-    // Sanity check: previously used memory mapped file should be closed before opening a new
-    // one
-    if (m_memory_mapped_segment_file.is_open()) {
+    // Sanity check: previously used memory mapped file should be closed before opening a new one
+    if (m_memory_mapped_segment_file.has_value()) {
         SPDLOG_WARN(
                 "streaming_archive::reader::Segment: Previous segment should be closed before "
                 "opening new one: {}",
                 segment_path.c_str()
         );
-        m_memory_mapped_segment_file.close();
+        m_memory_mapped_segment_file.reset();
     }
-    // Create read only memory mapped file
-    boost::iostreams::mapped_file_params memory_map_params;
-    memory_map_params.path = segment_path;
-    memory_map_params.flags = boost::iostreams::mapped_file::readonly;
-    memory_map_params.length = segment_file_size;
-    // Try to map it to the same memory location as the previous memory mapped file
-    memory_map_params.hint = m_memory_mapped_segment_file.data();
-    m_memory_mapped_segment_file.open(memory_map_params);
-    if (!m_memory_mapped_segment_file.is_open()) {
+
+    // Create read-only memory mapped file
+    m_memory_mapped_segment_file.emplace(segment_path);
+    if (false == m_memory_mapped_segment_file.has_value()) {
```

Comment:
Shouldn't this logic be that we catch `ReadOnlyMemoryMappedFile::OperationFailed` and then print the log and return the error?

Reply:
right this is a major mistake... thanks for catching this issue. Solved in the latest commit